From 4619d76d36e04e64a6072305b7f6dc851059376b Mon Sep 17 00:00:00 2001
From: Arne Morten Kvarving <arne.morten.kvarving@sintef.no>
Date: Wed, 17 Oct 2018 12:55:31 +0200
Subject: [PATCH] fixed: all settings are retransferred on exit

add flag that we do not want to reinit during shutdown

--- a/src/Main.cpp
+++ b/src/Main.cpp
@@ -95,6 +95,7 @@ class ATTRIBUTE_HIDDEN CVisualizationProjectM
   int m_lastPresetIdx;
   unsigned int m_lastLoggedPresetIdx;
   bool m_lastLockStatus;
+  bool m_shutdown = false;
 
   // some projectm globals
   const static int maxSamples=512;
@@ -144,6 +145,7 @@ CVisualizationProjectM::~CVisualizationProjectM()
 {
   unsigned int lastindex = 0;
   m_projectM->selectedPresetIndex(lastindex);
+  m_shutdown = true;
   kodi::SetSettingInt("lastpresetidx", lastindex);
   kodi::SetSettingString("lastpresetfolder", m_projectM->settings().presetURL);
   kodi::SetSettingBoolean("lastlockedstatus", m_projectM->isPresetLocked());
@@ -299,7 +301,7 @@ ADDON_STATUS CVisualizationProjectM::SetSetting(const std::string& settingName,
   else if (settingName == "beat_sens")
     m_configPM.beatSensitivity = settingValue.GetInt() * 2;
 
-  if (settingName == "beat_sens") // becomes changed in future by a additional value on function
+  if (settingName == "beat_sens" && !m_shutdown) // becomes changed in future by a additional value on function
   {
     if (!InitProjectM())    //The last setting value is already set so we (re)initalize
       return ADDON_STATUS_UNKNOWN;
